### PR TITLE
[2.0] Allow for tests to be run on PHP 8

### PR DESCRIPTION
**Pull request by pierlon**
## Summary

This PR adds PHP 8 support for v2.0 by fixing the PHP 8 Travis job to ensure tests are passing and adds the `^8.0` constraint to `composer.json`.

All commits are cherry-picked from #5475.

## Checklist

- [ ] My pull request is addressing an open issue (please create one otherwise).
- [ ] My code is tested and passes existing [tests](https://github.com/ampproject/amp-wp/wiki/Engineering-Guidelines#tests).
- [ ] My code follows the [Engineering Guidelines](https://github.com/ampproject/amp-wp/wiki/Engineering-Guidelines) (updates are often made to the guidelines, check it out periodically).

### CODE REVIEW

**Comment by westonruter** [comment]
Test failures 😢

File: .travis.yml [hunk 1]
```diff
@@ -59,13 +59,9 @@ jobs:
...
   # These need to be exact matches, including whitespace!
   allow_failures:
-    # PHP unit tests (7.3, WordPress trunk)
-    - env: WP_VERSION=trunk  DEV_LIB_ONLY=phpunit                               INSTALL_PWA_PLUGIN=1
-    # PHP and JavaScript unit tests (7.3, WordPress trunk, with code coverage)
-    - env: WP_VERSION=latest DEV_LIB_ONLY=phpunit                               INSTALL_PWA_PLUGIN=1 RUN_PHPUNIT_COVERAGE=1
     # PHP unit tests (8.0, WordPress trunk)
     - env: WP_VERSION=trunk  DEV_LIB_ONLY=phpunit                               INSTALL_PWA_PLUGIN=1
-      php: nightly
+      php: "nightly"
```
Comment (swissspidy) [inline .travis.yml]: Should be 8.0?

Comment (pierlon) [reply .travis.yml]: I'm not sure to be honest, but I'll give it a try. According to this [thread](https://travis-ci.community/t/php-8-0-missing/10132/) `8.0snapshot` is available, but `8.0` isn't as yet due to [compilation issues](https://travis-ci.community/t/php-8-0-missing/10132/15).

**Comment by westonruter** [comment]
I'm confused why there are test failures in the common/optimizer libs since I thought they were pinned to a specific version of the Optimizer spec tests, and so https://github.com/ampproject/amp-toolbox-php/pull/39 shouldn't be needed here.

**Comment by pierlon** [comment]
@westonruter The Optimizer spec tests are updated whenever `composer update` is run, so that's why it's failing now.

**Comment by pierlon** [comment]
I've backported the fix from https://github.com/ampproject/amp-toolbox-php/pull/39 in https://github.com/ampproject/amp-wp/pull/5637/commits/21fb4868197363721a549960860b168494585975.

**Comment by westonruter** [comment]
Commit dc995ad is cherry-picked from PR #5654 for `develop`.

File: .travis.yml (hunk 1 above)
Comment (pierlon) [reply .travis.yml]: It seems the 8.0 image is now available.

**Comment by pierlon** [comment]
Seems the Travis PHP 8.0 image is now available, so I'm reverting https://github.com/ampproject/amp-wp/pull/5637/commits/75cb871c34bcb532ce2384039fd0be6d6dbb0354.

**Review by westonruter — APPROVED** [review]

**Comment by westonruter** [comment]
QA passed as PHP 8 tests are passing in 2.0 branch:

image

**Pull request merged**